packages/camera/camera_platform_interface/lib/src/events/device_event.dart

```diff
@@ -0,0 +1,47 @@
+import 'package:camera_platform_interface/src/utils/utils.dart';
+import 'package:flutter/services.dart';
+
+/// Generic Event coming from the native side of Camera,
+/// not related to a specific camera module.
+///
+/// This class is used as a base class for all the events that might be
+/// triggered from a device, but it is never used directly as an event type.
+///
+/// Do NOT instantiate new events like `DeviceEvent(cameraId)` directly,
+/// use a specific class instead:
+///
+/// Do `class NewEvent extend DeviceEvent` when creating your own events.
+/// See below for examples: `DeviceOrientationChangedEvent`...
+/// These events are more semantic and more pleasant to use than raw generics.
+/// They can be (and in fact, are) filtered by the `instanceof`-operator.
+abstract class DeviceEvent {}
+
+/// An event fired whenever.
+class DeviceOrientationChangedEvent extends DeviceEvent {
+  /// The new orientation the device
```

Comment:
nit: grammar mistake (missing the word "of"):

```suggestion
  /// The new orientation of the device
```